ironfish-cli/src/commands/service/snapshot.ts

```diff
@@ -0,0 +1,213 @@
+import { Assert, AsyncUtils, FileUtils, GENESIS_BLOCK_SEQUENCE } from '@ironfish/sdk'
+import { CliUx, Flags } from '@oclif/core'
+import { spawn } from 'child_process'
+import crypto from 'crypto'
+import fsAsync from 'fs/promises'
+import os from 'os'
+import path from 'path'
+import { parseNumber } from '../../args'
+import { IronfishCommand } from '../../command'
+import { RemoteFlags } from '../../flags'
+import { ProgressBar } from '../../types'
+
+export default class CreateSnapshot extends IronfishCommand {
+  static hidden = true
+
+  static description = `Upload chain snapshot to a public bucket`
+
+  static flags = {
+    ...RemoteFlags,
+    bucket: Flags.string({
+      char: 'e',
+      parse: (input: string) => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Bucket URL to upload snapshot to',
+    }),
+    path: Flags.string({
+      char: 'p',
+      parse: (input: string): Promise<string> => Promise.resolve(input.trim()),
+      required: false,
+      description: 'The path where the snapshot should be saved',
+    }),
+    maxBlocksPerChunk: Flags.integer({
+      char: 'm',
+      required: false,
+      default: isNaN(Number(process.env.MAX_BLOCKS_PER_SNAPSHOT_CHUNK))
+        ? 1000
+        : Number(process.env.MAX_BLOCKS_PER_SNAPSHOT_CHUNK),
+      description: 'The max number of blocks per file in the zipped snapshot',
+    }),
+  }
+
+  static args = [
+    {
+      name: 'start',
+      parse: (input: string): Promise<number | null> => Promise.resolve(parseNumber(input)),
+      default: Number(GENESIS_BLOCK_SEQUENCE),
+      required: false,
+      description: 'the sequence to start snapshot at (inclusive, genesis block is 1)',
+    },
+    {
+      name: 'stop',
+      parse: (input: string): Promise<number | null> => Promise.resolve(parseNumber(input)),
+      required: false,
+      description: 'the sequence to snapshot end at (inclusive)',
+    },
+  ]
+
+  async start(): Promise<void> {
+    const { flags, args } = await this.parse(CreateSnapshot)
+
+    const bucket = (flags.bucket || process.env.IRONFISH_SNAPSHOT_BUCKET || '').trim()
+    if (!bucket) {
+      this.log(
+        `Cannot upload snapshot without bucket URL. You must set IRONFISH_SNAPSHOT_BUCKET or pass --bucket flag.`,
+      )
+      this.exit(1)
+    }
+
+    let exportDir
+
+    if (flags.path) {
+      exportDir = this.sdk.fileSystem.resolve(flags.path)
+    } else {
+      try {
+        exportDir = await fsAsync.mkdtemp(`${os.tmpdir()}${path.sep}`)
+      } catch (err) {
+        this.log(`Could not create temp folder for snapshot generation`)
+        this.exit(1)
+      }
+    }
+    Assert.isNotUndefined(exportDir)
+
+    const blockExportPath = this.sdk.fileSystem.join(exportDir, 'blocks')
+    await this.sdk.fileSystem.mkdir(blockExportPath, { recursive: true })
+
+    this.log('Connecting to node...')
+
+    const client = await this.sdk.connectRpc()
```

Comment:
Have you tried passing `forceLocal` here so you will not attempt to connect to a currently running node? I think it then will crap out if you are trying to spin up a second node, and should help avoid the slowness issue.

Reply:
I changed the command to use the `forceLocal` flag and the command now waits for the node to close before exporting a snapshot.